spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarConfiguration.java

```diff
@@ -80,13 +97,29 @@ public EurekaInstanceConfigBean eurekaInstanceConfigBean() {
 			int port = this.sidecarProperties.getPort();
 			config.setNonSecurePort(port);
 			config.setInstanceId(getDefaultInstanceId(this.env));
-			if(StringUtils.hasText(springAppName)) {
+			if (StringUtils.hasText(springAppName)) {
 				config.setAppname(springAppName);
 				config.setVirtualHostName(springAppName);
 				config.setSecureVirtualHostName(springAppName);
 			}
-			if (StringUtils.hasText(this.hostname)) {
-				config.setHostname(this.hostname);
+			String hostname = this.sidecarProperties.getHostname();
+			String ipAddress = this.sidecarProperties.getIpAddress();
+			if (!StringUtils.hasText(hostname) && StringUtils.hasText(this.hostname)) {
+				hostname = this.hostname;
+			}
+			if (StringUtils.hasText(hostname)) {
+				if (!StringUtils.hasText(ipAddress)) {
+					try {
+						ipAddress = inetUtils.convertAddress(InetAddress.getByName(hostname)).getIpAddress();
```

Comment:
We already have code to do this [here](https://github.com/spring-cloud/spring-cloud-commons/blob/master/spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java).  It is already used in the constructor of [`EurekaInstanceConfigBean`](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaInstanceConfigBean.java#L291-L296). Please remove this code.